hail/python/test/hail/test_ir.py

```diff
@@ -570,3 +572,52 @@ def test_refs(self):
                     ' (bar (GetField idx (Ref row)))))'
         )
         assert expected == CSERenderer()(x)
+
+
+def _assert_encoding_roundtrip(value):
+    lit = hl.literal(value)
+    round_trip = lit.dtype._from_encoding(lit.dtype._to_encoding(value))
+    if isinstance(value, np.ndarray):
+        assert_array_equal(round_trip, value)
+    else:
+        assert round_trip == value
+
+
+@pytest.mark.parametrize(
+    'value',
+    [
+        1,
+        5.0,
+        "foo",
+        [1, 2, 3, 4],
+        (5, 6, 7, 8),
+        {"foo", "bar", "baz"},
+        {"a": 1, "b": 2},
+        np.array([[1, 2], [3, 4], [5, 6]]),
+        np.array([[1, 2], [3, 4], [5, 6]]).T,
+    ]
+)
+def test_python_value_literal_encodings(value):
+    _assert_encoding_roundtrip(value)
+
+
+def test_call_encoding():
+    calls = [
+        hl.Call([0, 1]),
+        hl.Call([1, 0], phased=True),
+        hl.Call([2], phased=True),
+        hl.Call([]),
+        hl.Call([1, 1]),
+        hl.Call([17495, 17495]),
+    ]
+    for call in calls:
+        _assert_encoding_roundtrip(call)
+
+
+def test_locus_interval_encoding():
+    start = hl.Locus(contig='chr1', position=10001, reference_genome='GRCh38')
+    end = hl.Locus(contig='chr1', position=11001, reference_genome='GRCh38')
+    interval = hl.Interval(start=start, end=end, includes_start=True, includes_end=False)
+    _assert_encoding_roundtrip(start)
+    _assert_encoding_roundtrip(end)
+    _assert_encoding_roundtrip(interval)
```

Comment:
Each of these three tests is slightly different, but they all ultimately call the same method. What's the argument against just parameterizing `_assert_encoding_roundtrip` itself?

Reply:
I tried that at first, and looks like I can do this with `Call`, but I can't instantiate a `Locus` before `init` (because it tries to load a reference genome) so I can't include a `Locus` in the parametrize.

Reply:
ahah right